File: Program12.java
/*
	Write a program which accept number from user and return multiplication of all digits.
*/

import java.util.Scanner;

class Digits
{

	private int iNo;
	
	public Digits(int iNum)
	{
		iNo = iNum;
	}
	
	public int multOfDigits()
	{
		int iDigit = 0,iMult = 1;
		
		if(iNo < 0)
		{
			iNo = -iNo;
		}
		
		while(iNo > 0)
		{
			iDigit = iNo % 10;
			
			if(iDigit == 0)
			{
				iDigit = 1;
			}	
			
			iMult = iMult * iDigit;
			
			iNo = iNo / 10;
			
		}
		return iMult;
	}


}

class Program12
{
	
	public static void main(java.lang.String Arg[])
	{
		int iValue = 0;
		
		Scanner sObj = new Scanner(System.in);
		
		System.out.println("Enter number : ");
		iValue = sObj.nextInt();
		
		Digits dObj = new Digits(iValue);
		
		System.out.println("Multiplication of all digits is : "+dObj.multOfDigits());
		
		sObj.close();	
	}
}

File: Program12.py

# Write a program which accept number from user and return multiplication of all digits.

class Digits:
	
	def __init__(self,iNum):
		
		self.iNo = iNum
	
	def multOfDigits(self):
		
		iMult = 1
		
		if(self.iNo < 0):
			
			self.iNo = -self.iNo
		
		while(self.iNo > 0):
				
			iDigit = self.iNo % 10
			
			if(iDigit == 0):
				
				iDigit = 1
				
			iMult = iMult * iDigit
			
			self.iNo //= 10
		
		return iMult
				

def main():
	
	print("Enter Number : ")
	iValue = int(input())
	
	dObj = Digits(iValue)
	
	print("\nMultiplication of all digits is : ",dObj.multOfDigits())

if __name__ == "__main__":
	main()
